src/mc_status.cpp

```diff
@@ -0,0 +1,69 @@
+#include "mc_status.h"
+
+#include "common.h"
+#include "mbed.h"
+
+#define MC_STATUS_TIMEOUT_MS (5000ms) // This will need some fine tuning
+
+static InterruptIn mc_status(MC_STATUS_PIN, PullNone);
+static Ticker      mc_status_ticker;
+static int         flash_counter  = 0;
+static mc_status_e current_status = MC_STATUS_OK;
+
+void mc_status_private_incrementCounter(void) { flash_counter++; }
+
+void mc_status_private_resetCounter(void) { flash_counter = 0; }
+
+void mc_status_private_computeStatus(void)
+{
+    if(flash_counter >= MC_STATUS_MAX_ERROR_CODE)
+    {
+        current_status = MC_STATUS_MAX_ERROR_CODE;
+    }
+    else
+    {
+        current_status = (mc_status_e)flash_counter;
+    }
+    mc_status_private_resetCounter();
+}
+
+bool mc_status_init()
+{
+    mc_status.rise(&mc_status_private_incrementCounter);
+    mc_status_ticker.attach(&mc_status_private_computeStatus,
+                            MC_STATUS_TIMEOUT_MS);
+    return true;
+}
+
+mc_status_e mc_status_getStatus() { return current_status; }
+
+char *mc_status_getStatusString(mc_status_e status)
+{
+    switch(status)
+    {
+    case MC_STATUS_OK:
+        return "OK";
+    case MC_STATUS_OVERCURRENT:
+        return "Overcurrent";
+    case MC_STATUS_NOT_SET:
+        return "Not set";
+    case MC_STATUS_HOLE_SENSOR_ERROR:
+        return "Hole sensor error";

```

Comment:
```suggestion
    case MC_STATUS_HALL_SENSOR_ERROR:
        return "Hall sensor error";
```

Reply:
no

Reply:
![MicrosoftTeams-image](https://github.com/badgerloop-software/sc1-mcc/assets/72111610/ba992a75-644d-4519-8d92-98871c97204b)

Reply:
![image](https://github.com/badgerloop-software/sc1-mcc/assets/8378883/03805d89-cf27-40b5-8c37-656639a375c2)
lmao

Reply:
![image](https://github.com/badgerloop-software/sc1-mcc/assets/8378883/050f3c71-c4e1-4a3e-ae08-f29831bbe4ba)

Marking as resolved...

Reply:
It'S a HOlE seNsOr

Reply:
Hole sensor? I hardly know 'er!